opentelemetry-sdk/src/metrics/in_memory_exporter.rs

```diff
@@ -4,19 +4,20 @@ use crate::metrics::data::{
 };
 use crate::metrics::exporter::PushMetricExporter;
 use crate::metrics::Temporality;
-use crate::InMemoryExporterError;
-use std::collections::VecDeque;
 use std::fmt;
 use std::sync::{Arc, Mutex};
 use std::time::Duration;
 
 use super::data::{AggregatedMetrics, Metric, ScopeMetrics};
 
+// Not a user-facing type, just a type alias for clarity within this module.
+type InMemoryMetrics = Vec<ResourceMetrics>;
+
 /// An in-memory metrics exporter that stores metrics data in memory.
 ///
 /// This exporter is useful for testing and debugging purposes. It stores
-/// metric data in a `VecDeque<ResourceMetrics>`. Metrics can be retrieved
-/// using the `get_finished_metrics` method.
+/// metric data in a user-provided `Vec<ResourceMetrics>`, from which the
+/// exported data can be retrieved as well.
```

Comment:
I think we should be more explicit here. Something like this perhaps?

```suggestion
  /// This exporter is useful for testing and debugging purposes. It stores
  /// metric data in a user-provided `Arc<Mutex<Vec<ResourceMetrics>>>`. Users
  /// retain a clone of this `Arc` and can access the exported metrics directly
  /// by locking the mutex.
```